src/algorithm/invert.rs

```diff
@@ -1280,56 +1280,30 @@ fn invert_dup_pattern<'a>(
 
     // As under's undo step
     for end in (1..=input.len()).rev() {
-        let instrs = &input[..end];
-        if instrs_clean_signature(instrs).map_or(true, |sig| sig != (1, 1)) {
+        let mut instrs = &input[..end];
+        let Some(Signature { args, outputs: 1 }) = instrs_clean_signature(instrs) else {
+            continue;
+        };
+        if !instrs_are_pure(instrs, &comp.asm, Purity::Pure) {
             continue;
         }
-        let Ok((_before, mut after)) = under_instrs(instrs, Signature::new(1, 1), comp) else {
+        let Ok((before, mut after)) = under_instrs(instrs, Signature::new(1, 1), comp) else {
             continue;
         };
+        println!("before: {before:?}\nafter: {after:?}");
```

Comment:
This line is causing the CI failure. One of the tests checks for leftover `println!`s and `dbg!`s.

Reply:
oh oops